packages/extension/src/serviceWorker/index.ts

```diff
@@ -516,8 +516,9 @@ chrome.debugger.onEvent.addListener((source, method, params) => {
       ? cookie.domain.slice(1)
       : cookie?.domain;
 
-    // Adding alternate domains here because our extension calculates domain differently that the application tab.
-    // This is done to capture both NID.google.com/ and NIDgoogle.com/ so that if we find either of the cookie we add issues to the cookie object
+    if (!domainToUse) {
+      return;
+    }
```

Comment:
I think this check can be an early return. it can be clubbed with the condition on line 504-507 or a additional check after that.